Project infra (#11)
squid-cache/squid-cache.github.io: docs/ProjectInfrastructure/DailySnapshotsGeneration.md

# How daily snapshots files are generated

As of the 2023-11-02, this is the process to generate snapshot tarballs.

1. Jenkins jobs run at regular cadences, monitoring the repository, run a full build test.

>    [trunk-arm64-matrix](https://build.squid-cache.org/job/trunk-arm64-matrix/) is used for master,

Reviewer: Lets be clear this is the VCS branch we use "master" later to mean the server.
```suggestion
   [`trunk-arm64-matrix`](https://build.squid-cache.org/job/trunk-arm64-matrix/) is used for the VCS ["master" branch](https://github.com/squid-cache/squid/tree/master/),
```

>    while `6-matrix` and `5-matrix` are (at this time) used for the stable and old releases

Reviewer: ```suggestion
   while a version specific `N-matrix` (e.g. `6-matrix` or `5-matrix`) are used for the stable and old releases
```

1. if these jobs are successful, they will trigger the corresponding tarball creation job

>    (e.g. [website-tarballs-head](https://build.squid-cache.org/job/website-tarballs-head/) for trunk).

Reviewer: ```suggestion
   (e.g. [`website-tarballs-head`](https://build.squid-cache.org/job/website-tarballs-head/) for the `trunk_amd64_matrix`).
```

>    The tarball jobs trust that if they are invoked, it's because the branch is stable.

Reviewer: "stable" is often misinterpreted as "production ready". Our meaning here is more specific, so lets be clear.

```suggestion
   The tarball jobs trust that if they are invoked, it's because the branch has a stable build and testing result.
```

>    Jenkins administrators can at any point force the execution of these jobs

Reviewer: punctuation typo.
```suggestion
   Jenkins administrators can, at any point, force the execution of these jobs.
```

> 1. The code run there is [make-snapshot.sh](https://github.com/kinkie/support-tools/blob/master/squid-ci/make-snapshot.sh).

Reviewer: ```suggestion
1. The code run there is [`make-snapshot.sh`](https://github.com/kinkie/support-tools/blob/master/squid-ci/make-snapshot.sh).
```

   The artifacts it generates are replicated to buildmaster and are accessible via https.

>    e.g. [master artifacts](https://build.squid-cache.org/job/website-tarballs-head/lastSuccessfulBuild/artifact/artifacts/).

Reviewer: ```suggestion
   e.g. [development version artifacts](https://build.squid-cache.org/job/website-tarballs-head/lastSuccessfulBuild/artifact/artifacts/).
```

> 1. A cron job running (mk-release-snapshots.sh) running on on master downloads these artifacts,

Reviewer: ```suggestion
1. A cron job running [`mk-release-snapshots.sh`](https://github.com/squid-cache/ci/blob/main/release/mk-release-snapshots.sh) on master downloads these artifacts,
```
